lib/gremlin.js

```diff
@@ -19,15 +20,24 @@ var java = require('java');
 
 var Gremlin = module.exports = function (opts) {
 
-  if (opts && opts.java === java) {
+  // java.onJvmCreated is a property that holds a function that is executed only once when
+  // the JVM has been created. After creation, the property is made inaccessible.
+  var jvmCreated = !_.isFunction(java.onJvmCreated);
+
+  if (jvmCreated) {
+    dlog('Gremlin initialized after JVM already created. Java:', java);
+  }
+  else if (opts && opts.java) {
```

Comment:
Actually, at no point has it been necessary to pass java in. A boolean 'applicationHasAlreadyConfiguredJava' would have served just as well.

But there are three cases worth distinguishing: 1) jvm already created, 2) jvm not yet created, but java configuration already done, and 3) jvm not yet created, and gremlin should do configuration. Knowing that the JVM has already been created is not enough to distinguish between the other two cases.

FYI, if the JVM has not been created, the creation will happen a few lines down when `java.newInstanceSync()` is called.

The 3rd option is the backwards compatible option. While working on these changes I noticed that in unit tests we create a Gremlin instance multiple times, even though the JVM is created only once, which is the main reason I added detection of JVM already created. I still want the ability to fully configure java outside of Gremlin, but I am open to changing the way that it is indicated.

However, let me point out that it will all be moot in a week or two, as we're about to eliminate the need for gremlin-v3. So, let me know if you really want me to make a change here. If you agree that it's moot, please just approve and merge.